src/cfnlint/data/AdditionalSpecs/SubNeededExcludes.json

```diff
@@ -63,23 +63,54 @@
                 }
             }
         },
+        "AWS::EC2::SpotFleet": {
+            "Metadata": {
+                "AWS::CloudFormation::Init": {
+                    "ExcludeAll": true
+                }
+            },
+            "Properties": {
+                "SpotFleetRequestConfigData": {
+                    "Properties": {
+                        "LaunchSpecifications": {
+                            "Properties": {
+                                "UserData": {
+                                    "ExcludeAll": true
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "AWS::EC2::LaunchTemplate": {
+            "Properties": {
+                "LaunchTemplateData": {
+                    "Properties": {
+                        "UserData": {
+                            "ExcludeAll": true
+                        }
+                    }
+                }
+            }
+        },
         "AWS::EMR::InstanceGroupConfig": {
             "Properties": {
                 "AutoScalingPolicy": {
                     "Properties": {
-                        "ScalingRule": {
+                        "Rules": {
```

Comment:
We need to follow the actual Json/Yaml path and not the spec path because those items don't always match.

Reply:
Good catch! Apologies, it wasn't straight forward to reverse some of the hard-coded excludes originally in the rule, and clearly I missed the property name not aligning with the sub property type.